### ISSUE TITLE
THIS IS A VIRUS

### ISSUE THREAD

**Issue by ducclucc5** [BP-Innovation/Bad-Piggies-Origin#9]
Do not download this, this is a virus

**Comment by ZachjuKamashi** [comment]
> Do not download this, this is a virus

? Did you check it in virustotal, or did you assume that because of windows defender? false positives have always been a thing. I've checked it and only unitycrashhandler32 shows up with 1 Trojan which I bet is a false positive. Either way the source is quite literally posted and you are free to check anything malicious in the code and post it here.

**Comment by Scarlaid** [comment]
wheres the virus part
![image](https://github.com/user-attachments/assets/b05c71ec-e106-462a-b50e-a6f07da8e672)

**Comment by ducclucc5** [comment]
> wheres the virus part
> ![image](https://private-user-images.githubusercontent.com/90851437/393874939-b05c71ec-e106-462a-b50e-a6f07da8e672.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3ODQ4ODcsIm5iZiI6MTczMzc4NDU4NywicGF0aCI6Ii85MDg1MTQzNy8zOTM4NzQ5MzktYjA1YzcxZWMtZTEwNi00NjJhLWI1MGUtYTZmMDdkYThlNjcyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA5VDIyNDk0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxYzFkZTk1NGM3YmM0OWRiOThkZmMzNzY3ZmRmNjUzMWQ3YzMwMWI3MTdmNTQ0OTg3YjIxOWY1NGI2ODlmYzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._k-7NLgjCAg5BTJdDZbC3u-GDvrki72Gw-6XmmD9_vc)

There is a such thing as undetected malware. These are especially easily hidden in Unity games. This is how I know it it. I opened it in process monitor in a virtual machine. You filter what badpiggies.exe does and you will see that it opens and modified a lot of registry keys and drops files it’s not meant to. It is an undetected malware, even if you run it through virustotal, it’ll return nothing. Unity games are usually undetected by antiviruses. And with what zach said, I did a dynamic analysis, like I said before. Even if the source is available, the built binary is malicious and even the source could have bad code hidden. I found a task in taskscheduler called runtimebroker.exe, this is a legit program but it’s in appdata, so it’s it imitating it. Please check your taskscheduler for it

**Comment by ZachjuKamashi** [comment]
> There is a such thing as undetected malware. 

Yes there are but I feel like it really isn't malicious. It's been almost 5 years since the release and I definitely would have noticed something wrong by now as I've literally ran it on my main machine where I've even intentionally disabled Windows updates because A they are annoying and B they break the system entirely. I'm pretty sure I still get security updates but I don't know where and how to really look whether it's working or not.

> These are especially easily hidden in Unity games. This is how I know it it. I opened it in process monitor in a virtual machine. You filter what badpiggies.exe does and you will see that it opens and modified a lot of registry keys and drops files it’s not meant to. It is an undetected malware, even if you run it through virustotal, it’ll return nothing. Unity games are usually undetected by antiviruses. And with what zach said, I did a dynamic analysis, like I said before. Even if the source is available, the built binary is malicious and even the source could have bad code hidden. I found a task in taskscheduler called runtimebroker.exe, this is a legit program but it’s in appdata, so it’s it imitating it. Please check your taskscheduler for it

Could you do further research regarding this? if you can find the old official rovio release and it does the exact same thing it might be false assumptions.

**Comment by Scarlaid** [comment]
> > wheres the virus part
> > ![image](https://private-user-images.githubusercontent.com/90851437/393874939-b05c71ec-e106-462a-b50e-a6f07da8e672.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3ODQ4ODcsIm5iZiI6MTczMzc4NDU4NywicGF0aCI6Ii85MDg1MTQzNy8zOTM4NzQ5MzktYjA1YzcxZWMtZTEwNi00NjJhLWI1MGUtYTZmMDdkYThlNjcyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA5VDIyNDk0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxYzFkZTk1NGM3YmM0OWRiOThkZmMzNzY3ZmRmNjUzMWQ3YzMwMWI3MTdmNTQ0OTg3YjIxOWY1NGI2ODlmYzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._k-7NLgjCAg5BTJdDZbC3u-GDvrki72Gw-6XmmD9_vc)
> 
> There is a such thing as undetected malware. These are especially easily hidden in Unity games. This is how I know it it. I opened it in process monitor in a virtual machine. You filter what badpiggies.exe does and you will see that it opens and modified a lot of registry keys and drops files it’s not meant to. It is an undetected malware, even if you run it through virustotal, it’ll return nothing. Unity games are usually undetected by antiviruses. And with what zach said, I did a dynamic analysis, like I said before. Even if the source is available, the built binary is malicious and even the source could have bad code hidden. I found a task in taskscheduler called runtimebroker.exe, this is a legit program but it’s in appdata, so it’s it imitating it. Please check your taskscheduler for it

I cannot consume "concrete" information without proper source
I have also run this on [triage](https://tria.ge/) first then run on my main device, to this day nothing was found, same as `runtimebroker.exe` is still the only thing inside `WinSxS` on my Windows 
![image](https://github.com/user-attachments/assets/aa752720-585e-4611-8a93-726579d33846)

I get the idea of being extra cautious but source code is there and can look inside it if wanted
For the official Bad Piggies PC port if you want to try, [here](https://archive.org/details/bad-piggies-full-collection)

**Comment by ducclucc5** [comment]
I’m quite new to the cyber security space, and was sus of it. I’m sorry if I’m wrong. I’ll try to do more tests, I may of linked a different malware to it. Sorry guys, I’ll test it again later.
